playwright/tests/common/environment.spec.ts

```diff
@@ -0,0 +1,205 @@
+import { expect } from "@playwright/test";
+
+import { test } from "../../fixtures";
+import { mockManagerProcedures } from "../../utils";
+import { environments } from "../../mocks";
+
+test.describe("Environment", () => {
+  test.run()(
+    "I can click on the text to log in",
+    async ({ sliceMachinePage }) => {
+      await sliceMachinePage.gotoDefaultPage();
+      await sliceMachinePage.environmentSelector.loginTextButton.click();
+      await expect(sliceMachinePage.loginDialog.submitButton).toBeVisible();
+    },
+  );
+
+  test.run()(
+    "I can click on the icon to log in",
+    async ({ sliceMachinePage }) => {
+      await sliceMachinePage.gotoDefaultPage();
+      await sliceMachinePage.environmentSelector.loginIconButton.click();
+      await expect(sliceMachinePage.loginDialog.submitButton).toBeVisible();
+    },
+  );
+
+  test.run()(
+    'I can see "Production" environment when no environments exist',
+    async ({ sliceMachinePage }) => {
+      await mockManagerProcedures({
```

Comment:
⚠️ Are you sure about your mocking here? I feel like "no env exist" means that either you have an empty array or no active env. Here you return "production", but what about the mock to active env? It could happen that a Slice Machine tested version have one.

Reply:
I guess can you could mock `fetchEnvironments` and `readEnvironment` to return nothing and so it should be "Production" as a fallback.

Reply:
The environment endpoint returns `prod`, `stage`, and `dev` environments, so the array should never be empty (it will always have at least the `prod` environment).

It is only treated as "empty" if the user is logged out since we are unable to fetch a list of the repository's environments.

Reply:
Hum and since you don't mock the active env, what if the active env is "stage" for example.

Reply:
This case led me down a larger refactor: https://github.com/prismicio/slice-machine/pull/1253

In the case where an invalid environment is selected, Slice Machine will revert to the production environment. The above PR has a test case for that scenario.